apps/publisher/src/hooks/useMediaDevices.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from "react";
+
+type MediaDevices = {
+    cameraList: InputDeviceInfo[];
+    microphoneList: InputDeviceInfo[];
+    setCameraHandler: (device: InputDeviceInfo) => void;
+    setMicrophoneHandler: (device: InputDeviceInfo) => void;
+    selectedCamera?: InputDeviceInfo;
+    selectedMicrophone?: InputDeviceInfo;
+    mediaStream: MediaStream | undefined;
+}
+
+const useMediaDevices: () => MediaDevices = () => {
+    const [cameraList, setCameraList] = useState<InputDeviceInfo[]>([]);
+    const [microphoneList, setMicrophoneList] = useState<InputDeviceInfo[]>([]);
+    
+    const [selectedCamera, setCamera] = useState<InputDeviceInfo>();
```

Comment:
I am thinking we should separate `select` which is actually from user, `set` is react state change.
So I suggest here
```suggestion
    const [camera, setCamera] = useState<InputDeviceInfo>();
```